src/main/java/seedu/address/logic/parser/FindContactParser.java

```diff
@@ -0,0 +1,37 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import seedu.address.logic.commands.FindContactCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.contact.NameContainsKeywordsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindContactCommand object
+ */
+public class FindContactParser implements Parser<FindContactCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindContactCommand
+     * and returns a FindContactCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindContactCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindContactCommand.MESSAGE_USAGE));
+        }
+
+        String[] nameKeywords = trimmedArgs.split("\\s+");
+
+        List<String> keywords = new ArrayList<>(Arrays.asList(nameKeywords));
+        NameContainsKeywordsPredicate predicate = new NameContainsKeywordsPredicate(keywords);
```

Comment:
I like how this is separated into 2 lines (not combined into 1 line like the original code) which can enhance readability 😄